scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,401 @@
+        current_status = None
+
+        ring_id_regex = re.compile(r"^(?:RING|LINK) ID (\d+)")
+        id_ip_regex = re.compile(r"(?:id|addr)= ([\d\.]+)")
+        status_regex = re.compile(r"^status+= (.+)$")
+        node_status_regex = re.compile(r"^node (\d+)\: link enabled\:\d+ link connected\:\d+")
+
+        for line in lines:
+            ring_match = ring_id_regex.match(line)
+            if ring_match:
+                current_ring = ring_match.group(1)
+
+            old_status_match = status_regex.match(line)
+            new_status_match = node_status_regex.match(line)
+            if old_status_match:
+                current_status = old_status_match.group(1)
+            elif new_status_match:
+                
+            id_ip_match = id_ip_regex.match(line)
+            if id_ip_match:
+                current_ip = id_ip_match.group(1)
+
+            if current_ring and current_status and current_ip:
+                status_check = "active with no faults" in current_status
+                rings_info[current_ring] = {
+                    "ip": current_ip,
+                    "status_text": current_status,
+                    "status_check": status_check,
+                }
+                current_ring = current_status = current_ip = None
+
+        return rings_info
+
+    @staticmethod
+    def check_if_service_is_active(service_name: str):
+        """
+        service_name for example : pacemaker
+        """
+        command = SYSTEMCTL_IS_ACTIVE_COMMAND.format(service_name)
+        _, stdout = StandbyNodeHealthChecker._run_command(command)
+        if stdout == "active":
+            print(f"Service {service_name} is active")
+            return True
+        print(f"Service {service_name} is not active")
+        return False
+
+    def get_ufm_ha_cluster_status_string(self):
+        command = UFM_HA_CLUSTER_COMMAND.format("status")
+        ret_code, res_string = StandbyNodeHealthChecker._run_command(command)
+        if ret_code != 0:
+            print("Failed to get ha cluster status")
+            res_string = ""
+        self._ufm_ha_status_string = res_string
+
+    def _get_status_value_from_ha_cluster_status(self, keyword: str):
+        for line in self._ufm_ha_status_string.splitlines():
+            line = line.strip()
+            if line.startswith(keyword):
+                return line.split(":", 1)[-1].strip().lower()
+        return None
+
+    def check_drbd_connectivity(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drbd_connectivity_status = self._get_status_value_from_ha_cluster_status(
+            "DRBD_CONNECTIVITY"
+        )
+        if drbd_connectivity_status != "[connected]":
+            print(
+                f"DRBD CONNECTIVITY status is {drbd_connectivity_status} and not Connected"
+            )
+            return False
+        print("DRBD CONNECTIVITY status is ok (Connected)")
+        return True
+
+    def check_drdb_resource(self):
```

Comment:
you don't need to check resource. need to check role

Reply:
Fixed